geoutils/raster/interpolate.py

```diff
@@ -0,0 +1,329 @@
+        interp_mask = RegularGridInterpolator(
+            coords, new_mask, method="nearest", bounds_error=bounds_error, fill_value=1
+        )
+
+        # We replace all NaN values by nearest neighbours to minimize interpolation errors near NaNs
+        # Elegant solution from: https://stackoverflow.com/questions/5551286/filling-gaps-in-a-numpy-array
+        indices = distance_transform_edt(mask_nan, return_distances=False, return_indices=True)
+        values = values[tuple(indices)]
+
+    # For the RegularGridInterpolator
+    if method in RegularGridInterpolator._ALL_METHODS:
+
+        # We create the classic interpolator
+        interp = RegularGridInterpolator(
+            coords, values, method=method, bounds_error=bounds_error, fill_value=fill_value
+        )
+
+        # We create a new interpolator callable that propagates nodata as defined above
+        def regulargrid_interpolator_with_nan(xi: tuple[NDArrayNum, NDArrayNum]) -> NDArrayNum:
+
+            results = interp(xi)
+
+            # If NaNs are not supported
+            if True:
+                invalids = interp_mask(xi)
+                results[invalids.astype(bool)] = np.nan
+
+            return results
+
+        return regulargrid_interpolator_with_nan
+
+    # For the RectBivariateSpline
+    else:
+
+        # The coordinates must be in ascending order, which requires flipping the array too (more costly)
+        interp = RectBivariateSpline(np.flip(coords[0]), coords[1], np.flip(values[:], axis=0))
+
+        # We create a new interpolator callable that propagates nodata as defined above, and supports fill_value
+        def rectbivariate_interpolator_with_fillvalue(xi: tuple[NDArrayNum, NDArrayNum]) -> NDArrayNum:
+
+            # Get invalids
+            invalids = interp_mask(xi)
+
+            # RectBivariateSpline doesn't support fill_value, so we need to wrap here to add them
+            xi_arr = np.array(xi).T
+            xi_shape = xi_arr.shape
+            xi_arr = xi_arr.reshape(-1, xi_arr.shape[-1])
+            idx_valid = np.all(
+                (
+                    coords[0][-1] <= xi_arr[:, 0],
+                    xi_arr[:, 0] <= coords[0][0],
+                    coords[1][0] <= xi_arr[:, 1],
+                    xi_arr[:, 1] <= coords[1][-1],
+                ),
+                axis=0,
+            )
+            # Make a copy of values for RectBivariateSpline
+            result = np.empty_like(xi_arr[:, 0])
+            result[idx_valid] = interp.ev(xi_arr[idx_valid, 0], xi_arr[idx_valid, 1])
+            result[np.logical_not(idx_valid)] = fill_value
+
+            # Add back NaNs from dilated mask
+            results = np.atleast_1d(result.reshape(xi_shape[:-1]))
+            results[invalids.astype(bool)] = np.nan
+
+            return results
+
+        return rectbivariate_interpolator_with_fillvalue
+
+
+def _map_coordinates_nodata_propag(
+    values: NDArrayNum,
```

Comment:
same comment about arguments not described in docstring

Reply:
Same as above: pointing towards `scipy.ndimage.map_coordinates` as they are the same arguments 1:1.